src/date.ts

```diff
@@ -0,0 +1,151 @@
+import 'globalize/dist/globalize/date';
+import 'globalize/dist/globalize/relative-time';
+import getGlobalize, { normalizeOptions } from './util/globalize';
+
+export type DateFormatterOptions = null | {
+	/**
+	 * String value indicating a skeleton, eg. { skeleton: "GyMMMd" }.
+	 * Skeleton provides a more flexible formatting mechanism than the predefined list full, long, medium, or short represented by date, time, or datetime.
+	 * Instead, they are an open-ended list of patterns containing only date field information, and in a canonical order.
+	 */
+	skeleton?: string;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { date: "full" }.
+	 */
+	date?: DateLength;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { time: "full" }.
+	 */
+	time?: DateLength;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { datetime: "full" }
+	 */
+	datetime?: DateLength;
+}
+
+export type DateLength = 'short' | 'medium' | 'long' | 'full';
+
+export type RelativeTimeFormatterOptions = null | {
+	/**
+	 * eg. "short" or "narrow". Or falsy for default long form
+	 */
+	form?: RelativeTimeLength;
+}
+
+export type RelativeTimeLength = 'short' | 'narrow';
+
+/**
+ * Format a date according to the specified options for the specified or current locale.
+ *
+ * @param value
+ * The date to format.
+ *
+ * @param options
+ * An optional object of formatting options.
+ *
+ * @param locale
+ * An optional locale. Defaults to the root locale.
+ *
+ * @return
+ * The formatted date string.
+ */
+export function formatDate(value: Date, options?: DateFormatterOptions, locale?: string): string {
```

Comment:
I don't know how nice this would be but you could overload the function and internally support the second argument being the locale (perhaps)?

```ts
export function formatDate(value: Date, options?: DateFormatterOptions, locale?: string): string;
export function formatDate(value: Date, locale?: DateFormatterOptions): string;
export function formatDate(value: Date, optionsOrLocale?: DateFormatterOptions | string, locale?: string): string {
    if (typeof optionsOrLocale === 'string') {
    ...
}

```